internal/pkg/service/common/configpatch/dump_test.go

```diff
@@ -0,0 +1,288 @@
+package configpatch_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/configpatch"
+)
+
+type Config struct {
+	Key1 string        `json:"foo1"`
+	Key2 int           `json:"foo2" alternative:"baz2" protected:"true"`
+	Key3 ConfigNested1 `json:"foo3,omitempty" alternative:"baz3"`
+}
+
+type ConfigNested1 struct {
+	Key4 string        `json:"foo4"`
+	Key5 int           `json:"foo5" alternative:"baz5"`
+	Key6 ConfigNested2 `json:"foo6,omitempty"`
+}
+
+type ConfigNested2 struct {
+	Key7 []string `json:"foo7"`
+	Key8 bool     `json:"foo8" protected:"true"`
+}
+
+type ConfigPatch struct {
+	Key1 *string             `json:"foo1"`
+	Key2 *int                `json:"foo2" alternative:"baz2"`
+	Key3 *ConfigNested1Patch `json:"foo3,omitempty" alternative:"baz3"`
+}
+
+type ConfigNested1Patch struct {
+	Key5 *int                `json:"foo5" alternative:"baz5"`
+	Key6 *ConfigNested2Patch `json:"foo6,omitempty"`
+}
+
+type ConfigNested2Patch struct {
+	Key7 *[]string `json:"foo7"`
+	Key8 *bool     `json:"foo8"`
+}
+
+type ConfigPatchInvalid struct {
+	Key1 string  `json:"foo1"`
+	Key2 *string `json:"foo2"`
+	Key8 *string `json:"foo8"`
+	Key9 *int    `json:"foo9"`
+}
+
+func TestDumpKVs_EmptyPatch(t *testing.T) {
+	t.Parallel()
+
```

Comment:
Here are the boring but necessary tests for `confipatch.ToKVs` function.

The code is fully covered, except for one panic.
![image](https://github.com/keboola/keboola-as-code/assets/19371734/fb5291c2-cb4f-4cde-bb58-3de8ce0b99eb)